Destination/RotatableStreamDestination.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Zenstruck\BackupBundle\Destination;
+
+use Psr\Log\LoggerInterface;
+use Zenstruck\BackupBundle\Rotator\Rotator;
+
+class RotatableStreamDestination extends StreamDestination
+{
+    /**
+     * @var null|Rotator
+     */
+    private $preRotator;
+
+    /**
+     * @var null|Rotator
+     */
+    private $postRotator;
+
+    /**
+     * A constructor
+     *
+     * @param string $directory Path to backup directory.
+     * @param Rotator|null $preRotator Rotator to execute prior to push.
+     * @param Rotator|null $postRotator Rotator to execute after push.
+     */
+    public function __construct($directory, Rotator $preRotator = null, Rotator $postRotator = null)
```

Comment:
I got the idea that you can provide instance of Rotator interface. If you have to use collection of rotators, there is a RotatorCollection that implements Rotator interface.